packages/library/src/commonMain/kotlin/io/realm/internal/RealmObjectHelper.kt

```diff
@@ -99,7 +100,18 @@ object RealmObjectHelper {
         //  RealmObjectInterop and make cinterop operate on primitive values and native pointers
         //  only. This relates to the overall concern of having a generic path for getter/setter
         //  instead of generating a typed path for each type.
-        RealmInterop.realm_set_value(o, key, value, false)
+        try {
+            RealmInterop.realm_set_value(o, key, value, false)
+        }
+        // The catch block should catch specific Core exceptions and rethrow them as Kotlin exceptions.
+        // Core exceptions meaning might differ depending on the context, by rethrowing we can add some context related
+        // info that might help users to understand the exception.
+        catch (exception: RealmCoreException) {
+            throw IllegalStateException(
```

Comment:
This still seems to throw IllegalState unconditionally? Shouldn't it test for `RLM_ERR_NOT_IN_A_TRANSACTION` or is this done in another PR?